src/t8_vtk/t8_vtk_reader.cxx

```diff
@@ -298,13 +302,108 @@ t8_vtk_iterate_cells (vtkSmartPointer < vtkDataSet > vtkGrid,
     T8_FREE (tuples);
   }
   T8_FREE (vertices);
-  return tree_id;
+  return;
+}
+
+/**
+ * Set the partition for cmesh coming from a non-distributed vtkGrid. 
+ * 
+ * \param[in, out] cmesh  The cmesh, uncommitted, partition set after call of this function 
+ * \param[in] mpirank     The mpirank of this proc.
+ * \param[in] main_proc   The rank of the main process.
+ * \param[in] num_trees   The number of trees.
+ * \param[in] dim         The dimension of the cmesh.
+ * \param[in] comm        The communicator to use. 
+ */
+static void
+t8_vtk_cmesh_partition (t8_cmesh_t cmesh, const int mpirank,
+                        const int main_proc, t8_gloidx_t num_trees, int dim,
+                        sc_MPI_Comm comm)
+{
+  t8_gloidx_t         first_tree;
+  t8_gloidx_t         last_tree;
+  if (mpirank == main_proc) {
+    first_tree = 0;
+    last_tree = num_trees - 1;
+  }
+  /* Communicate the dimension to all processes */
+  sc_MPI_Bcast (&dim, 1, sc_MPI_INT, main_proc, comm);
```

Comment:
In case we have a single reading process the other processes will have an empty cmesh.  The cmesh needs to set the correct dimension because it can't be changed afterwards. Therefore we need to communicate it. If the process does not know the dim the input can be arbitrary. Currently it is set to 0.

Reply:
Understood.